articles/golang-part-2-programming-basics/index.md

```diff
@@ -0,0 +1,304 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: golang-part-2-programming-basics
+title:  Golang - Programming Basics
+description: It covers the basics of programming in Go: directories, workspaces, variables, loops, conditionals and control flow, etc.
+author: adith-bharadwaj
+date: 2020-07-21T00:00:00-07:00
+topics: [languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/golang-part-2-programming-basics/hero.jpg
+    alt: golang image example
+---
+
+*Go is a statically-typed and procedural programming language whose syntax resembles C*. In the previous article, we looked at the history of Go, its purpose, and installation. In this article, we are going to explore basic programming concepts such as variables, input/output, arrays, etc. Before we start writing programs, there are a few concepts to bear in mind.
+<!--more-->
+
+### Go workspace
+*Go programs are usually kept in the same workspace*. A workspace may contain multiple repositories from version control systems such as Git.  
+These repositories can contain several packages. According to the [documentation](https://golang.org/), Go programs are organized into packages. A package is a collection of programs in the same directory that are compiled together.
+
+*A repository contains one or more modules*. A module is a collection of related Go packages that are released together. A Go repository typically contains only one module, located at the root of the repository.
+
+#### Directories
+A workspace is a directory hierarchy with two directories at its root: src, bin, and pkg.
+
+**src**: *The src directory contains the source code (Go programs) written by the user. It may contain multiple repositories*.
+
+**bin**: *This folder contains the binary executable files that are built and installed by the Go tool*.
+
+**pkg**: *This directory contains all the non-executable packages or shared libraries that are used in other executable packages*.
+
+This is an example of a workspace that contains an executable program called hello inside the bin directory and multiple programs inside the src directory in a hierarchical structure.
+
+```
+bin/
+    hello                                                   
+src/
+    github.com/golang/example/                     
+        hello/
+            hello.go
+        program/
+            program.go               
+```
+
+### Program structure
+Go programs have the following structure:
+
+```go
+package main
+
+import "fmt"
+
+func main() {
+
+   fmt.Println("Hello, World!")
+}
+```
+
+1. Package declaration: *This is The first line of the program and defines the name of the current package name. "Package main", tells Go that this is the starting point to run the program. We can use the **package** keyword to define packages in our program*.
```

Comment:
"Package main"
Should this be 
"package main" ?

Reply:
might look better using code formatting by wrapping with the ` symbol

`package main`